src/main/java/htsjdk/samtools/SBIIndexWriter.java

```diff
@@ -0,0 +1,150 @@
+package htsjdk.samtools;
+
+import htsjdk.samtools.util.BinaryCodec;
+import htsjdk.samtools.util.IOUtil;
+import htsjdk.samtools.util.RuntimeIOException;
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * Writes SBI files as understood by {@link SBIIndex}.
+ *
+ * <p>To use this class, first construct an instance from an output stream, and a desired
+ * granularity. Then for each record in the file being indexed, pass the virtual file offset of the
+ * record to the {@link #processRecord} method. The indexer will keep a count of the records passed
+ * in an index every <i>n</i>th record. When there are no records left call {@link #finish} to
+ * complete writing the index.
+ */
+public final class SBIIndexWriter {
```

Comment:
If this will be in htsjdk, add a TODO pointing to the PR to be removed once htsjdk is upgraded with that change.